src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -177,6 +238,155 @@ public class PatientEverythingService : IPatientEverythingService
             return new SearchResult(searchResultEntries, searchResult.ContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
         }
 
+        private void CheckForReplacedByLinks(string parentPatientId, SearchResultEntry parentPatientResource)
+        {
+            List<Patient.LinkComponent> links = ExtractLinksFromParentPatient(parentPatientResource);
+
+            if (links == null)
+            {
+                return;
+            }
+
+            // If a "replaced-by" link is present, it indicates that the patient resource containing this link must no
+            // longer be used. The link points forward to another patient resource that must be used in lieu of the
+            // resource that contains the "replaced-by" link.
+            foreach (Patient.LinkComponent link in links)
+            {
+                // Regardless of the other links present, we throw an error if we find a "replaced-by" link.
+                if (link.Type == Patient.LinkType.ReplacedBy)
+                {
+                    ReferenceSearchValue referenceSearchValue = _referenceSearchValueParser.Parse(link.Other.Reference);
+
+                    // Specify the url that must be used instead.
+                    Uri url = _urlResolver.ResolveOperationResultUrl(OperationsConstants.PatientEverything, referenceSearchValue.ResourceId);
+
+                    throw new EverythingOperationException(
+                        string.Format(
+                            Core.Resources.EverythingOperationResourceIrrelevant,
+                            parentPatientId,
+                            referenceSearchValue.ResourceId),
+                        HttpStatusCode.MovedPermanently,
+                        url.ToString());
+                }
+            }
+        }
+
+        private List<Patient.LinkComponent> ExtractLinksFromParentPatient(SearchResultEntry parentPatientResource)
+        {
+            // If it wasn't found, it means the parent patient doesn't exist in the database.
+            if (parentPatientResource.Resource == null)
+            {
+                // There are no links to extract since there is no parent patient resource.
+                return null;
+            }
+
+            // Otherwise, we found the parent patient in the database. Extract its links.
+            ResourceElement element = _resourceDeserializer.Deserialize(parentPatientResource.Resource);
+            Patient parentPatient = element.ToPoco<Patient>();
+
+            return parentPatient.Link;
+        }
+
+        private async Task<string> CheckForNextSeeAlsoLinkAndSetToken(string parentPatientId, EverythingOperationContinuationToken token, CancellationToken cancellationToken)
+        {
+            // Retrieve the parent patient so that we can extract its links and process the next "seealso" link.
+            using IScoped<ISearchService> search = _searchServiceFactory();
```

Comment:
nit, prefer the following which more clearly indicates its a delegate:
```suggestion
            using IScoped<ISearchService> search = _searchServiceFactory.Invoke();
```